tests/conftest.py

```diff
@@ -74,6 +80,45 @@ def signing_credentials_fixture() -> Credentials:  # The same account is generat
     )
 
 
+@dataclass
+class DevnetAccount:
+    address: str
+    private_key: str
+    public_key: str
+    signer: StarkCurveSigner
+
+
+@pytest.fixture(name="devnet_accounts")
+def devnet_accounts_fixture(devnet_gateway_url: str) -> list[DevnetAccount]:
+    response = requests.get(f"{devnet_gateway_url}/predeployed_accounts")
+    devnet_account_dicts = json.loads(response.content)
+    return [
+        DevnetAccount(
+            address=devnet_account_dict["address"],
+            private_key=devnet_account_dict["private_key"],
+            public_key=devnet_account_dict["public_key"],
+            signer=StarkCurveSigner(
+                account_address=devnet_account_dict["address"],
+                key_pair=KeyPair(
+                    private_key=int(devnet_account_dict["private_key"], base=16),
+                    public_key=int(devnet_account_dict["public_key"], base=16),
+                ),
+                chain_id=StarknetChainId.TESTNET,
+            ),
+        )
+        for devnet_account_dict in devnet_account_dicts
+    ]
+
+
+@pytest.fixture(name="alice_devnet_account")
+def alice_devnet_account(
+    devnet_accounts: list[DevnetAccount], monkeypatch: pytest.MonkeyPatch
+) -> DevnetAccount:
+    alice_devnet_account = devnet_accounts[0]
+    monkeypatch.setenv(PRIVATE_KEY_ENV_VAR_NAME, alice_devnet_account.private_key)
```

Comment:
this is an unwanted side effect of the fixture

Reply:
The same can be said about running the devnet when using `devnet_gateway_url` fixture. I'll make it explicit.

Reply:
Done in [8bebc31](https://github.com/software-mansion/protostar/pull/881/commits/8bebc31924e5c8c24462a97146fc7ca29e401bef)